back/src/Kyoo.Meilisearch/FilterExtensionMethods.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.ComponentModel.DataAnnotations;
+using Kyoo.Abstractions.Models.Utils;
+using static System.Text.Json.JsonNamingPolicy;
+
+namespace Kyoo.Meiliseach;
+
+internal static class FilterExtensionMethods
+{
+	public static string? CreateMeilisearchFilter<T>(this Filter<T>? filter)
+	{
+		return filter switch
+		{
+			Filter<T>.And and
+				=> $"({and.First.CreateMeilisearchFilter()}) AND ({and.Second.CreateMeilisearchFilter()})",
+			Filter<T>.Or or
+				=> $"({or.First.CreateMeilisearchFilter()}) OR ({or.Second.CreateMeilisearchFilter()})",
+			Filter<T>.Gt gt => CreateBasicFilterString(gt.Property, ">", gt.Value),
+			Filter<T>.Lt lt => CreateBasicFilterString(lt.Property, "<", lt.Value),
+			Filter<T>.Ge ge => CreateBasicFilterString(ge.Property, ">=", ge.Value),
+			Filter<T>.Le le => CreateBasicFilterString(le.Property, "<=", le.Value),
+			Filter<T>.Eq eq => CreateBasicFilterString(eq.Property, "=", eq.Value),
+			Filter<T>.Has has => CreateBasicFilterString(has.Property, "=", has.Value),
+			Filter<T>.Ne ne => CreateBasicFilterString(ne.Property, "!=", ne.Value),
+			Filter<T>.Not not => $"NOT ({not.Filter.CreateMeilisearchFilter()})",
+			Filter<T>.CmpRandom
+				=> throw new ValidationException("Random comparison is not supported."),
+			_ => null
+		};
+	}
+
+	private static string CreateBasicFilterString(string property, string @operator, object? value)
+	{
+		return $"{CamelCase.ConvertName(property)} {@operator} {value.InMeilisearchFormat()}";
+	}
+
+	private static object? InMeilsearchFilterFormat(this object? value)
+	{
+		return value switch
+		{
+			null => null,
+			string s => s.Any(char.IsWhiteSpace) ? $"\"{s}\"" : s,
```

Comment:
String containing `"` would fail there, I think they should be escaped (I have not checked the meili documentation for how to escape).

Reply:
I was following the documentation where it says to wrap strings with spaces in them with either double or single quotes: https://www.meilisearch.com/docs/learn/fine_tuning_results/filtering#examples

But, to be fair I haven't had a chance to test this because the number of properties I can filter on are limited to `kind, status, airDate, startAir, endAir, studioId, genres` so we could just remove this handing and add it if/when it's required.

Reply:
Sorry, I didn't read your comment correctly the first time. Yes they might need to be escaped. I might just remove this for now. We can add it if/when we have properties we can filter on that have spaces in the text.

Reply:
I'd rather leave it on, the API would already work with every filter that way.

I just checked in their issue tracker, we can escape `"` with `\`.

Reply:
Ok I've added escape chars 👌